rules/ast/functions.go

```diff
@@ -498,6 +499,43 @@ func derivImpl(timestamp clientmodel.Timestamp, args []Node) interface{} {
 	return resultVector
 }
 
+// === histogram_quantile(k ScalarNode, vector VectorNode) Vector ===
```

Comment:
> That confuses me.

`histogram_quantile(.9, rate(histo[5m]))` is an error as you'd have (e.g.) two timeseries with a le=1.0 label.

`histogram_quantile(.9, sum by (le)(rate(histo[5m])))` would be the correct way to use histogram_quantile without a by clause.